src/profiler.ts

```diff
@@ -162,10 +163,8 @@ export class Profiler extends common.ServiceObject {
       json: true,
     };
 
-    return this.request(options).then((result) => {
-      const [body, response] = result;
-      return body;
-    });
+    const [body, response] = await this.request(options);
```

Comment:
What's the intention around how errors in the request will be dealt with?

Reply:
Here, an error should be thrown for certain types of errors (I need to check what we do with our other profilers to be certain which types of errors) which we know won't be resolved going forward. Otherwise, profile creation should be retried with exponential back-off, capped at one hour. 
I'll implement the logic for backing off on retries and checking which errors should be retried in a later PR.